97.md

```diff
@@ -0,0 +1,82 @@
+NIP-97
+======
+
+Event Recommendation Services
+-----------------------------
+
+`draft` `optional` `author:vitorpamplona`
+
+## Abstract
+
+Service providers want to offer event recommendations to users that subscribe to them. Specifics of the recommendation algorithms will differ between event kinds and provider implementations. This NIP describes a general framework for subscribing and performing recommendation requests.
+
+# Recommendation Subscriptions list
+
+A special event with kind `10020` "Recommendation Subscription List" is defined as a [NIP-16 - Replaceable Events](16.md) of public and/or private `p` tags, one for each of the subscribed services the author intends to use. Each tag entry should contain the public key for the service provider and a relay URL where the provider hosts the recommendation events. i.e., `["p", <32-bytes hex key>, <main relay URL>]`.
+
+For example:
+
+```json
+{
+  "kind": 10020,
+  "tags": [
+    ["p", "91cf9..4e5ca", "wss://provider1.com/"],
+    ["p", "14aeb..8dad4", "wss://provider2.com/nostr"],
+    ["p", "612ae..e610f", "ws://provider3.com/ws"]
+  ],
+  "content": "<private list>",
+  ...other fields
+}
+```
+
+Providers can offer as many algorithms as they way as long as they offer one per public key. 
+
+# Recommendation Request
+
+Clients can request recommendations at will using a kind `20020` ephemeral event. `p` tags MUST contain the group of users to make the recommendation for as well as the pub key of the service provider. The `.content` field MUST contains an array of NOSTR `REQ` filters ([NIP-1 - Basic Protocol](1.md)) that describe the type and constraints of the requested recommendation events.
+
+```json
+{
+  "kind": 20020,
+  "tags": [
+    ["p", "<service pubkey>"],
+    ["p", "91cf9..4e5ca", "wss://relay1.com"],
+  ],
+  "content": "[<FILTER1>, <FILTER2>, ...]",
+  ...other fields
+}
+```
+
+Clients may specify several recommendation filters, i.e. `[{ "kinds": [1, 1065]}, { "#p": ["91cf9..4e5ca"], limit 200}, ...]`. Clients may include `kinds`, `tags` and other filter fields to restrict recommendation results to particular event kinds.
+
+Providers SHOULD interpret the recommendation request to the best of their ability and return events that it deems interesting to the requesting pubkey group. Recommendations can also include trending events as long as they are deemed of interest.
```

Comment:
It's good that you mention trending here, bcs 'recommendation' can be 'personalized' or 'global' (and probably many more types too). You're probably ok merging trending stuff from several providers, and personalized stuff from several providers, but mixing 'trending' with 'personalized' will be confusing. So a user either has to specify several different recommendation subscription lists (say one for trending, one for personalized), which means it should become 30k event, or we should be able to add some standardized labels of what kind of recommendation is being requested. I'd prefer the first option - let users evaluate recommendation algos and add them to different lists and let them switch btw those lists.

Reply:
Agreed.

Reply:
So should the subscription list become 30xxx kind, so that users could have several of them?